File: myPcCovid/app/src/main/java/com/giangnb/pc_covid_clone/features/splash/SplashActivity.java
package com.giangnb.pc_covid_clone.features.splash;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.giangnb.pc_covid_clone.databinding.ActivitySplashBinding;
import com.giangnb.pc_covid_clone.features.login.LoginActivity;

public class SplashActivity extends AppCompatActivity {
    private ActivitySplashBinding binding;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySplashBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        Handler handler = new Handler();
        Runnable runnable = new Runnable() {
            @Override
            public void run() {
                startActivity(new Intent(SplashActivity.this, LoginActivity.class));
                finish();
            }
        };
        handler.postDelayed(runnable,3000);

    }
}

File: vnCalling/app/src/main/java/com/android/vncalling/ui/features/splash/SplashActivity.kt
package com.android.vncalling.ui.features.splash

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Handler
import android.os.Looper
import androidx.lifecycle.ViewModelProvider
import com.android.vncalling.base.BaseActivity
import com.android.vncalling.databinding.ActivitySplashBinding
import com.android.vncalling.ui.features.container.MainActivity
import com.android.vncalling.ui.features.login.LoginActivity
import com.android.vncalling.ui.features.welcome.WelcomeActivity
import com.android.vncalling.utils.Constants

@SuppressLint("CustomSplashScreen")
class SplashActivity : BaseActivity<ActivitySplashBinding, SplashViewModel>() {

    private val handler: Handler = Handler(Looper.getMainLooper())
    private val runnable: Runnable = Runnable {
        when (viewModel.decideNextActivity()) {
            Constants.WELCOME_ACTIVITY -> {
                val intent = Intent(this, WelcomeActivity::class.java)
                startActivity(intent)
                finish()
            }
            Constants.LOGIN_ACTIVITY -> {
                val intent = Intent(this, LoginActivity::class.java)
                startActivity(intent)
                finish()
            }
            Constants.MAIN_ACTIVITY -> {
                val intent = Intent(this, MainActivity::class.java)
                startActivity(intent)
                finish()
            }
        }
    }

    override fun createViewModel(): SplashViewModel {
        return ViewModelProvider(this)[SplashViewModel::class.java]
    }

    override fun getViewBinding(): ActivitySplashBinding {
        return ActivitySplashBinding.inflate(layoutInflater)
    }

    override fun initialize() {
        this.handler.postDelayed(runnable, 1200)
    }
}